### PR TITLE
ec2-api: Add SSL support

### PR DESCRIPTION
**Pull request by s-t-e-v-e-n-k**
Add support to deploy ec2-api with SSL enabled endpoints, optionally
generating self-signed certificates.

Co-Authored-By: Steve Kowalik <steven@wedontsleep.org>

This replaces https://github.com/crowbar/crowbar-openstack/pull/773 .

### CODE REVIEW

**Comment by nicolasbock** [comment]
@s-t-e-v-e-n-k I should have asked whether you tried to deploy this. Have you?

**Comment by s-t-e-v-e-n-k** [comment]
I have successfully deployed this on vp4. The reason for my branch over Sayali's is that this one won't behave unexpectedly if you set ec2-api to HTTPS and nova to HTTP (although, why you would is beyond me)

**Comment by nicolasbock** [comment]
@s-t-e-v-e-n-k thanks!

**Comment by dirkmueller** [comment]
@s-t-e-v-e-n-k I would have preferred to continue the discussion in the original PR instead of forking it, and thereby loosing all review history..

File: chef/cookbooks/ec2-api/recipes/ec2api.rb [hunk 1]
```diff
@@ -23,10 +23,9 @@
 
 # NOTE: ec2 is deployed via the nova barclamp
 ha_enabled  = node[:nova]["ec2-api"][:ha][:enabled]
-# SSL support is for the entire barclamp
-ssl_enabled = node[:nova][:ssl][:enabled]
```
Comment (dirkmueller) [inline chef/cookbooks/ec2-api/recipes/ec2api.rb]: This is used in other places within this file as well, and that means you've now changed ssl enablement from the global flag to the ec2-api flag. that should be reverted.

File: chef/data_bags/crowbar/template-nova.json [hunk 2]
```diff
@@ -42,6 +42,15 @@
...
           "database": "ec2"
+        },
+        "ssl": {
+          "enabled": false,
+          "certfile": "/etc/ec2api/ssl/certs/signing_cert.pem",
+          "keyfile": "/etc/ec2api/ssl/certs/signing_key.pem",
+          "generate_certs": false,
+          "insecure": false,
+          "cert_required": false,
+          "ca_certs": "/etc/barbican/ssl/certs/ca.pem"
```
Comment (dirkmueller) [inline chef/data_bags/crowbar/template-nova.json]: barbican?

File: chef/cookbooks/ec2-api/recipes/ec2api.rb (hunk 1 above)
Comment (s-t-e-v-e-n-k) [inline chef/cookbooks/ec2-api/recipes/ec2api.rb]: Global setting? ec2-api can not use nova's ssl certificates due to file permissions, so why should it make use of nova's ssl setting, when we could potentially be in the situation where nova is deployed  with SSL enabled and ec2-api is not.

File: chef/data_bags/crowbar/template-nova.json (hunk 2 above)
Comment (s-t-e-v-e-n-k) [inline chef/data_bags/crowbar/template-nova.json]: Good catch.

File: chef/cookbooks/ec2-api/recipes/ec2api.rb (hunk 1 above)
Comment (dirkmueller) [inline chef/cookbooks/ec2-api/recipes/ec2api.rb]: indeed, this behavior change is actually making it better, retracting my comment

**Comment by dirkmueller** [comment]
[(5) S57P2 S56P9 (S55P8) (S51P11 S50P12 S49P14) Re-add Nova EC2 API that got removed in Mitaka and added as new project](https://trello.com/c/x51hX6aZ/269-5-s57p2-s56p9-s55p8-s51p11-s50p12-s49p14-re-add-nova-ec2-api-that-got-removed-in-mitaka-and-added-as-new-project)

**Pull request merged**